networkit/cpp/embedding/BiasedRandomWalk.cpp

```diff
@@ -34,10 +34,31 @@ BiasedRandomWalk::BiasedRandomWalk(const Graph *graph)
     // pre-allocate unordered maps:
     graph->forNodes([&](node v) {
         auto degv = graph->degreeOut(v);
-        for (auto n : graph->neighborRange(v)) {
-            // init index2node:
-            index2node[v].push_back(n);
-            graphData->data[v][n] = AliasSampler(degv);
+        if (graph->isDirected()) {
+            // local numbering of neighbors by index in index2node[v]:
+            index2node[v].reserve(graph->degree(v));
+            index2node[v].insert(index2node[v].begin(), graph->neighborRange(v).begin(),
+                                 graph->neighborRange(v).end());
+
+            if (graph->degreeIn(v) == 0) {
+                graphData->data[v][none] = AliasSampler(degv);
+                return;
+            }
+
+            for (auto t : graph->inNeighborRange(v)) {
+                // one AliasSampler with entries for all v-neighbors
+                // per every node t leading to v
+                graphData->data[v][t] = AliasSampler(degv);
+            }
+        } else {
+            index2node[v].reserve(graph->degree(v));
+            auto index = 0;
```

Comment:
Careful here that `index` in NetworKit is also an alias type for `uint64_t`. Also, it does not seem to be used.

Reply:
Hm, pushed the wrong version. Thanks for pointing out.